samcli/commands/deploy/command.py

```diff
@@ -150,11 +151,14 @@
     "the output AWS CloudFormation template. YAML is used by default.",
 )
 @click.option(
-    "--resolve-s3",
+    "--resolve-s3", required=False, is_flag=True, help="Automatically resolve s3 bucket for non-guided deployments."
+)
+@click.option(
+    "--resolve-image-repos",
     required=False,
     is_flag=True,
-    help="Automatically resolve s3 bucket for non-guided deployments."
-    "Do not use --s3-guided parameter with this option.",
+    help="Automatically create and delete ECR repos for image based functions in non-guided deployments."
+    "Auto created image repos will be deleted if the corresponding functions are removed.",
```

Comment:
I haven't seen "repo" being used in any AWS docs (compare `"repo" site:docs.aws.amazon.com` versus `"repository" site:docs.aws.amazon.com`).

```suggestion
    help="Automatically create and delete ECR repositories for image-based functions in non-guided deployments. "
    "Automatically created image repositories will be deleted if the corresponding functions are removed.",
```